continuous_integration/travis_install.sh

```diff
@@ -0,0 +1,135 @@
+    #sudo ln -sf /usr/lib/x86_64-linux-gnu/libxcb-render-util.so.0 /usr/lib/libxcb-render-util.so.0 
+
+    conda config --add channels dsdale24;
+    conda create -q -n test-environment python=$PY_VERSION sphinx sip qt5 pyqt5;
+    ldd "$HOME/miniconda/envs/test-environment/lib/qt5/plugins/platforms/libqxcb.so"
+
+    # libxcb-atom1-dev libxcb-event1-dev libxcb-icccm1-dev
+    # ldd /home/goanpeca/anaconda/envs/test-environment/lib/qt5/plugins/platforms/libqxcb.so
+    #	
+  elif [ "$USE_QT_API" = "PyQt4" ]; then
+    conda create -q -n test-environment python=$PY_VERSION sphinx sip qt pyqt;
+  elif [ "$USE_QT_API" = "PySide" ]; then
+    conda create -q -n test-environment python=$PY_VERSION sphinx sip qt pyside;
+  fi
+
+  conda install -q -n test-environment $TEST_PACKAGES
+}
+
+install_pyside()
+{
+  # Currently support for python 2.7, 3.3, 3.4
+  # http://stackoverflow.com/questions/24489588/how-can-i-install-pyside-on-travis
+  sudo apt-get install libqt4-dev;
+  pip install --upgrade pip;
+  pip install PySide --no-index --find-links=$WHEELHOUSE_URI;  
+  # Travis CI servers use virtualenvs, so we need to finish the install by the following
+  POSTINSTALL=$(find ~/virtualenv/ -type f -name "pyside_postinstall.py";)
+  python $POSTINSTALL -install;
+}
+
+install_qt4()
+{
+  # Install Qt and then update the Matplotlib settings
+  sudo apt-get install -q libqt4-dev pyqt4-dev-tools;
+
+  if [[ $PY_VERSION == 2.7* ]]; then
+    sudo apt-get install python-dev
+    sudo apt-get install -q python-qt4
+  else  
+    sudo apt-get install python3-dev
+  fi
+
+  # http://stackoverflow.com/a/9716100
+  LIBS=( PyQt4 sip.so )
+
+  VAR=( $(which -a python$PY_VERSION) )
+
+  GET_PYTHON_LIB_CMD="from distutils.sysconfig import get_python_lib; print (get_python_lib())"
+  LIB_VIRTUALENV_PATH=$(python -c "$GET_PYTHON_LIB_CMD")
+  LIB_SYSTEM_PATH=$(${VAR[-1]} -c "$GET_PYTHON_LIB_CMD")
+
+  for LIB in ${LIBS[@]}
+  do
+    sudo ln -sf $LIB_SYSTEM_PATH/$LIB $LIB_VIRTUALENV_PATH/$LIB
+  done
+}
+
+install_qt5()
+{
+  echo "Not supported yet"
+}
+
+
+install_apt_pip()
+{  
+  # Test for different Qt bindings
+  if [ "$USE_QT_API" = "PyQt5" ]; then
+    #sudo apt-get install -qq python-sip python-qt5 python-sphinx --fix-missing;
+    #sudo apt-get install -qq python3-sip python3-pyqt5 --fix-missing;
+    install_qt5
+  elif [ "$USE_QT_API" = "PyQt4" ]; then
+    install_qt4;
+  elif [ "$USE_QT_API" = "PySide" ]; then
+    install_pyside;
+  fi
+  
+  if [ "$PY_VERSION" = "2.7" ]; then
+    TEST_PACKAGES+=" rope"
+  fi
+  pip install -U $TEST_PACKAGES
+}
+
+if [ "$USE_CONDA" = true ] ; then
+  export TEST_PACKAGES="ipython-qtconsole matplotlib pandas pep8 psutil pyflakes pygments pylint sphinx sympy"
+  export SOURCE=`source activate test-environment`
+  install_conda;
+else
+  export TEST_PACKAGES="IPython jedi matplotlib pandas pep8 psutil pyflakes pygments pylint sphinx sympy"
```

Comment:
1. Travis uses environments (with virtual env), so using apt-get will install packages for the global python which is 2.7 and that makes no sense for python 3.3, and 3.4. Travis VMs are all the same, what changes is the virtual envs....
2. Yes, we can use wheels (if they are available...) for all the other packages that take too long too compile. Otherwise we would need to make our own wheels for the different packages we are testing.